src/core/evaluator.js

```diff
@@ -1909,18 +1941,14 @@ class PartialEvaluator {
             if (!shading) {
               throw new FormatError("No shading object found");
             }
-
-            var shadingFill = Pattern.parseShading(
+            const patternId = self.parsePattern({
+              keyObj: shading,
               shading,
-              null,
-              xref,
               resources,
-              self.handler,
-              self._pdfFunctionFactory,
-              localColorSpaceCache
-            );
-            var patternIR = shadingFill.getIR();
-            args = [patternIR];
+              localColorSpaceCache,
+              localPatternCache,
+            });
+            args = [patternId];
```

Comment:
As mentioned above, you probably want to add a `operatorList.addDependency(patternId);` before this line.